### PR TITLE
Better velocity obs output solving issue #74

### PR DESCRIPTION
**Pull request by bearecinos**
Hi, 

I did some quick changes to run_inv.py and model.py so we can write cloud point velocities as fenics_ice variable output. This will make plots more consistent as now the observations will be in the same coordinates of the mesh. Also there is no point of outputting composite velocities if we wont be using them anymore for anything.

solving issue #74 
Let me know if the changes are ok or if I made poor decisions on variable names etc...

I also added some extra lines to output the first guess of beta as I also need this for plotting the input data.

PS: sorry for the many commits had to solved conflicts and pulled from main/EdiGlacUQ 

Cheers

### CODE REVIEW

File: fenics_ice/model.py [hunk 1]
```diff
@@ -362,6 +374,31 @@ def interpolate(values, vtx, wts):
...
+        self.v_cloud_M = Function(self.M, name="v_obs_cloud", static=True)
+
+        # Fill via interpolation
+        self.u_cloud_Q.vector()[:] = interpolate(self.vel_obs['u_obs'], vtx_Q_c, wts_Q_c)
+        self.v_cloud_Q.vector()[:] = interpolate(self.vel_obs['v_obs'], vtx_Q_c, wts_Q_c)
+        self.u_std_cloud_Q.vector()[:] = interpolate(self.vel_obs['u_std'], vtx_Q_c, wts_Q_c)
+        self.v_std_cloud_Q.vector()[:] = interpolate(self.vel_obs['v_std'], vtx_Q_c, wts_Q_c)
+
+        self.u_cloud_M.vector()[:] = interpolate(self.vel_obs['u_obs'], vtx_M_c, wts_M_c)
+        self.v_cloud_M.vector()[:] = interpolate(self.vel_obs['v_obs'], vtx_M_c, wts_M_c)
```
Comment (jrmaddison) [inline fenics_ice/model.py:400]: Should different fields be interpolated here? Are `u_cloud_M` and `v_cloud_M` needed?

File: runs/run_inv.py [hunk 2]
```diff
@@ -129,18 +129,23 @@ def run_inv(config_file):
...
 
-        U_obs = project((mdl.v_obs_Q**2 + mdl.u_obs_Q**2)**(1.0/2.0), mdl.M)
-        U_obs.rename("uv_obs", "")
-        inout.write_variable(U_obs, params, name="uv_obs", outdir=diag_dir,
+        U_obs = project((mdl.v_cloud_Q**2 + mdl.u_cloud_Q**2)**(1.0/2.0), mdl.M)
+        U_obs.rename("uv_cloud", "")
+        inout.write_variable(U_obs, params, name="uv_cloud", outdir=diag_dir,
+                             phase_name=phase_name, phase_suffix=phase_suffix)
+
+        U_obs_std = project((mdl.v_std_cloud_Q ** 2 + mdl.u_std_cloud_Q ** 2) ** (1.0 / 2.0), mdl.M)
```
Comment (jrmaddison) [inline runs/run_inv.py:146]: I don't think this is the (projection of) the standard deviation of the speed. Perhaps exclude this from the output?

File: runs/run_inv.py [hunk 3]
```diff
@@ -129,18 +129,23 @@ def run_inv(config_file):
...
+        U_obs_std.rename("uv_cloud_std", "")
+        inout.write_variable(U_obs, params, name="uv_cloud_std", outdir=diag_dir,
```
Comment (jrmaddison) [inline runs/run_inv.py:148]: `U_obs` written out again

**Comment by jrmaddison** [comment]
> sorry for the many commits had to solved conflicts and pulled from main/EdiGlacUQ

You can squash merge if you want to combine

File: fenics_ice/model.py (hunk 1 above)
Comment (bearecinos) [reply fenics_ice/model.py:400]: good point only `u_cloud_Q` gets written I will delete this

File: fenics_ice/model.py [hunk 4]
```diff
@@ -362,6 +374,31 @@ def interpolate(values, vtx, wts):
...
+        vtx_M_c, wts_M_c = interp_weights(self.vel_obs['uv_obs_pts'],
+                                      M_coords)
+
+        # Define new functions to hold results
+        self.u_cloud_Q = Function(self.Q, name="u_obs_cloud", static=True)
+        self.v_cloud_Q = Function(self.Q, name="v_obs_cloud", static=True)
+        self.u_std_cloud_Q = Function(self.Q, name="u_std_cloud", static=True)
+        self.v_std_cloud_Q = Function(self.Q, name="v_std_cloud", static=True)
+        # There is no mask for this data!
+        self.u_cloud_M = Function(self.M, name="u_obs_cloud", static=True)
```
[the remark this answers is not in the document]
Comment (bearecinos) [reply fenics_ice/model.py]: I deleted all _M as we dont use it

File: runs/run_inv.py (hunk 2 above)
Comment (bearecinos) [reply runs/run_inv.py]: deleted

File: runs/run_inv.py (hunk 3 above)
Comment (bearecinos) [reply runs/run_inv.py]: deleted this output

**Review by jrmaddison — APPROVED** [review]

**Pull request merged**